### PR TITLE
ref(js): Cleanup notification/utils.tsx

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Move documentation comments above the function so LSP correctly picks
   up on them

 - Reduce some nesting with guard statements

### CODE REVIEW

**Review by mgaeta — APPROVED** [review]
Oops it looks like we we've been doing python style docstring this whole time 😬

File: static/app/views/settings/account/notifications/utils.tsx [hunk 1]
```diff
@@ -385,8 +384,6 @@ export const getParentField = (
     parentId: string
   ) => NotificationSettingsObject
 ): FieldObject => {
-  /** Render each parent and add a default option to the the field choices. */
```
Comment (billyvg) [inline static/app/views/settings/account/notifications/utils.tsx:388]: Can we auto format this?

**Review by billyvg — APPROVED** [review]

File: static/app/views/settings/account/notifications/utils.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply static/app/views/settings/account/notifications/utils.tsx:388]: What are you thinking?

Comment (billyvg) [reply static/app/views/settings/account/notifications/utils.tsx:388]: idk if there's a lint rule for it

Comment (EvanPurkhiser) [reply static/app/views/settings/account/notifications/utils.tsx:388]: Like no comments like `/** ... */`?

Or also auto wrapping?

Comment (billyvg) [reply static/app/views/settings/account/notifications/utils.tsx:388]: Comment style

**Pull request merged**